File: linkdownloader_video.py
import requests
from bs4 import BeautifulSoup
# archive_url="http://dl.upfdl.com/files/Series/Person%20of%20Interest/S01/"#"http://dl.upfdl.com/files/Series/Person%20of%20Interest/S01/480p/"#"http://dl.upfdl.com/files/Series/Person%20of%20Interest/S01/720p%20x265/"
def get_video_links(url):
    r=requests.get(url)
    soup= BeautifulSoup(r.content,"html.parser")

    links=soup.findAll('a')
    # print(links)
    video_links = [url + link.get('href') for link in links if link.get('href').endswith('mkv')]
    video_links= video_links+[url + link.get('href') for link in links if link.get('href').endswith('mp4')]
    # for link in video_links:
    #     print(link)

    other_links=[]
    other_links = [url + link.get('href') for link in links if (not link.get('href').endswith('mkv') and not link.get('href').endswith('mp4') and not link.get('href').endswith('../') )]
    # print(other_links)
    for otherlink in  other_links:
        print("----------------",otherlink,"---------------------")
        for link in video_links:
            print(link)

        print("==")
        video_links=video_links+get_video_links(otherlink)
        

    return video_links

def download_video_series(video_links): 
    for link in video_links: 
        file_name = link.split('/')[-1]
        print ("Downloading file:%s",file_name)
        r = requests.get(link, stream = True)

        with open(file_name, 'wb') as f: 
            for chunk in r.iter_content(chunk_size = 1024*1024): 
                if chunk: 
                    f.write(chunk) 
        print ("%s downloaded!\n"%file_name)
    
    print ("All videos downloaded!")
    return
def download():
    archive_url=document.getElementById("archive_link").innerHTML
    # getting all video links 
    video_links = get_video_links(archive_url) 
  
    # download all videos 
    download_video_series(video_links)   

document.getElementById("download").addEventListener("click", download())     



    

File: extension/linkdownloader_video.js
var video_links=[];
// var other_links=[];
// var links=[];
var downloadCompleted=0;
function scrapeAndDownload(url,newDocument)//scrapes given  html document object for furthur links.
{
    //console.log("here7");
    var reference= newDocument.getElementsByTagName("a");
    for(let i=0;i<reference.length;i++){
        if(reference[i].href===url)reference.splice(i,1);
        if (reference[i].href.endsWith(".mkv")||reference[i].href.endsWith(".mp4"))
        {
            // console.log(reference[i].href);
            // console.log("here13");
            // downloadLink(reference[i].href);
            video_links.push(reference[i].href);
            
        }
    }
    
    let indexToDownload=0;
    if(isleftToDownload(indexToDownload))
    {
        downloadLink(indexToDownload);
    }
        
    //document.getElementById("status").innerHTML="Finished Downloading "
}

    // let file=
    // let stream = IO.newOutputStream(file, "binary");




worker = new Worker('worker.js');

// console.log("here");
// document.getElementById("download").onclick="download()";     
$(document).ready(function(){
    $("#download").click(function(){
        //worker.postMessage('from main: download started')
        let url=getArchiveLink();
        retrieveAndDownloadVideos(url);
    });
  });


function getArchiveLink()
{
    //var url="https://www.google.com"//'dl.upfdl.com/files/Series/Person%20of%20Interest/S01/480p/';
    var url=document.getElementById("archive_link").value;
    return url;

}

function retrieveAndDownloadVideos(url){

    let xhr=new XMLHttpRequest;
    
//   console.log("unsent: "+xhr.status);
    xhr.open('GET',url);
    xhr.responseType="document";
//   console.log(url+"  opened: "+xhr.status);
    xhr.onprogress=function(event){
    document.getElementById("status").style.display="inline";
    
    document.getElementById("status").innerHTML="Downloading";
    //   console.log("loading: "+xhr.status);
    console.log("loaded- "+event.loaded);
    console.log("contentLength- "+event.total);
    };
    
    xhr.onload=function(){
        document.getElementById("status").innerHTML="Downloading Complete";
        console.log("done"+xhr.status);
        let newDocument=xhr.response;
        scrapeAndDownload(url,newDocument);
    //document.getElementById("content").innerHTML=xhr.response;
    //document.getElementById("content").style.display="inline";
    };
    xhr.send();
   
}  